onnx/defs/schema.h

```diff
@@ -21,6 +21,12 @@
 
 namespace ONNX_NAMESPACE {
 
+struct Node;
+struct Dimension;
+typedef std::vector<std::vector<Dimension>> NodeOutSizes;
+typedef std::vector<ONNX_NAMESPACE::TensorProto_DataType> NodeOutTypes;
+typedef std::function<void(const Node*, NodeOutTypes&, NodeOutSizes&)> InferenceFunction;
```

Comment:
Decouple the shape inference function (which is part of op definition) with IR please.